mixins/rtl-mixin.js

```diff
@@ -13,7 +13,7 @@ export const RtlMixin = dedupeMixin(superclass => class extends superclass {
 		const dir = document.documentElement.getAttribute('dir');
 		// avoid reflecting "ltr" for better performance
 		if (dir && dir !== 'ltr') {
-			this._dir = dir;
+			this.setAttribute('dir', 'rtl');
```

Comment:
So I was kind of hoping we could remove the `properties()` section altogether and then switch the places referencing `this._dir` to just reference `this.dir` instead.

Reply:
I think there are certain scenarios where modifying the DOM from within the constructor can cause a JS error.  I'm trying to remember where I first ran into this, and I think Allan ran into it as well.  🤔

Reply:
I think it was assigning `id` from within the constructor - `createElement()` would fail.  That was one of the reasons (if not _the_ reason) for the constructor tests that we later created.  I don't remember Allan's case - it might have been the same, but different component.

Reply:
You are exactly right Dave B. It's failing at this line: https://github.com/BrightspaceUI/core/blob/main/components/menu/menu.js#L167, since document.createElement does not allow for attributes 😢 .

Reply:
What if we kept `dir` as a property above and right here set `this.dir = 'rtl';` (so the property gets set immediately). Then the attribute reflection code in Lit should handle the rest.